src/toniarts/openkeeper/game/sound/SoundCategory.java

```diff
@@ -0,0 +1,140 @@
+ * along with OpenKeeper.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package toniarts.openkeeper.game.sound;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import toniarts.openkeeper.tools.convert.sound.BankMapFile;
+import toniarts.openkeeper.tools.convert.sound.SdtFile;
+import toniarts.openkeeper.tools.convert.sound.sfx.SfxGroupEntry;
+import toniarts.openkeeper.tools.convert.sound.sfx.SfxMapFileEntry;
+import toniarts.openkeeper.tools.convert.sound.sfx.SfxMapFile;
+import toniarts.openkeeper.utils.PathUtils;
+
+/**
+ *
+ * @author archdemon
+ */
+public class SoundCategory {
+    // default sounds categories
+    public static final String SPEECH_MENTOR = "SPEECH_MENTOR";
+    public static final String SPEECH_MULTIPLAYER = "SPEECH_MULTIPLAYER";
+    public static final String SPEECH_HORNY = "SPEECH_HORNY";
+    // TODO need to add some GUI sounds categories
+    private final String folder;
+    private final String name;
+    private final Map<Integer, SoundGroup> groups = new HashMap<>();
+
+    private static final Logger LOGGER = Logger.getLogger(SoundCategory.class.getName());
+
+    public SoundCategory(String name, boolean useGlobal) {
+        if (name == null || name.isEmpty()) {
+            throw new RuntimeException("Sound category is empty");
+        }
+
+        if (useGlobal) {
+            this.folder = PathUtils.DKII_SFX_GLOBAL_FOLDER;
+        } else {
+            this.folder = PathUtils.DKII_SFX_FOLDER + name.toLowerCase() + File.separator;
+        }
+
+        this.name = name;
+        parseGroups();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Map<Integer, SoundGroup> getGroups() {
+        return groups;
+    }
+
+    public SoundGroup getGroup(int id) {
+        return groups.get(id);
+    }
+
+    public SoundGroup getGroup(SoundGroup.SoundType type) {
+        return groups.get(type.getValue());
+    }
+
+    public SfxMapFile getSfxMapFile() {
+        File f = new File(PathUtils.getDKIIFolder() + folder
+                + name.toLowerCase() + "SFX.map");
+        if (f.exists()) {
+            return new SfxMapFile(f);
+        }
+
+        throw new RuntimeException("Sfx file of category " + name + " not exits");
+    }
+
+    public BankMapFile getBankMapFile() {
+        File f = new File(PathUtils.getDKIIFolder() + folder
+                + name.toLowerCase() + "BANK.map");
+        if (f.exists()) {
+            return new BankMapFile(f);
+        }
+
+        throw new RuntimeException("Bank file of category " + name + " not exits");
+    }
+
+    /**
+     * Get SdtFile
+     *
+     * @param archiveFilename name in BankMapFile
+     * @return SdtFile
+     */
+    @Nullable
+    public static SdtFile getSdtFile(String archiveFilename) {
+
+        for (String part : new String[] {"HD.sdt", "HD.SDT", "HW.sdt", "HW.SDT"}) {
```

Comment:
I never understood the difference between HD & HW... Now we prefer HD files, is that just a guess?